src/services/accounts/AccountsStakingPayoutsService.ts

```diff
@@ -533,29 +540,56 @@ export class AccountsStakingPayoutsService extends AbstractService {
 			era: EraIndex,
 			stakers: KeysAndExposures,
 			validatorIndex: ValidatorIndex,
+			validatorsOverviewEntries?: [StorageKey, Option<SpStakingPagedExposureMetadata>][],
 		): IAdjustedDeriveEraExposure {
 			const nominators: DeriveEraNominatorExposure = {};
 			const validators: DeriveEraValidatorExposure = {};
+			const validatorsOverview: Record<string, Option<SpStakingPagedExposureMetadata>> = {};
 
 			stakers.forEach(([key, exposure]): void => {
 				const validatorId = key.args[1].toString();
 
+				if (validatorsOverviewEntries) {
+					for (const validator of validatorsOverviewEntries) {
+						const validatorKey: StorageKey = validator[0];
+						const valKey: [string, string] = validatorKey.toHuman() as unknown as [string, string];
+						if (valKey) {
+							if (valKey[1].toString() === validatorId) {
+								validatorsOverview[validatorId] = validator[1];
+								break;
+							}
+						}
+					}
+				}
+
 				validators[validatorId] = exposure;
 
-				exposure.others.forEach(({ who }, validatorIndex): void => {
-					const nominatorId = who.toString();
+				if (exposure.others) {
+					exposure.others.forEach(({ who }, validatorIndex): void => {
+						const nominatorId = who.toString();
 
-					nominators[nominatorId] = nominators[nominatorId] || [];
-					nominators[nominatorId].push({ validatorId, validatorIndex });
-				});
+						nominators[nominatorId] = nominators[nominatorId] || [];
+						nominators[nominatorId].push({ validatorId, validatorIndex });
+					});
+				} else {
+					(exposure as unknown as Option<SpStakingExposurePage>)
```

Comment:
Might be safe to also check `isSome` on the `Option` before unwrapping it.

Reply:
Added checks with `isSome` in the code part mentioned here, as well as in 3 other parts of the code (in the `extractExposure` function) where I am also unwrapping. The cases though in the `extractExposure` is not adding much value since `unwrap` is also called later on these variables ([here](https://github.com/paritytech/substrate-api-sidecar/blob/80571f840f1f62789a9b2ee0c41486ae2c46c24d/src/services/accounts/AccountsStakingPayoutsService.ts#L429) and [here](https://github.com/paritytech/substrate-api-sidecar/blob/80571f840f1f62789a9b2ee0c41486ae2c46c24d/src/services/accounts/AccountsStakingPayoutsService.ts#L430)) but I prefer not to refactor that part right now if that is ok.